services/osparc-gateway-server/src/osparc_gateway_server/backend/osparc.py

```diff
@@ -182,3 +205,132 @@ async def do_check_workers(self, workers: List[Worker]) -> List[bool]:
+                cluster_name,
+                count,
+                max_workers,
+            )
+            count = max_workers
+
+        if count != cluster.count:
+            cluster_update["count"] = count
+
+        created_workers = []
+        submitted_workers = []
+        target_updates = []
+        newly_running = []
+        close_expected = []
+        for worker in cluster.workers.values():
+            if worker.status >= JobStatus.STOPPED:
+                continue
+            elif worker.name in closing_workers:
+                if worker.status < JobStatus.RUNNING:
+                    newly_running.append(worker)
+                close_expected.append(worker)
+            elif worker.name in active_workers:
+                if worker.status < JobStatus.RUNNING:
+                    newly_running.append(worker)
+            elif worker.name in closed_workers:
+                target = (
+                    JobStatus.STOPPED if worker.close_expected else JobStatus.FAILED
+                )
+                target_updates.append((worker, {"target": target}))
+            else:
+                if worker.status == JobStatus.SUBMITTED:
+                    submitted_workers.append(worker)
+                else:
+                    assert worker.status == JobStatus.CREATED
+                    created_workers.append(worker)
+
+        n_pending = len(created_workers) + len(submitted_workers)
+        n_to_stop = len(active_workers) + n_pending - count
+        if n_to_stop > 0:
+            for w in islice(chain(created_workers, submitted_workers), n_to_stop):
+                target_updates.append((w, {"target": JobStatus.STOPPED}))
+
+        if cluster_update:
+            self.db.update_cluster(cluster, **cluster_update)
+            self.queue.put(cluster)
+
+        self.db.update_workers(target_updates)
+        for w, u in target_updates:
+            self.queue.put(w)
+
+        if newly_running:
+            # At least one worker successfully started, reset failure count
+            cluster.worker_start_failure_count = 0
+            self.db.update_workers(
+                [(w, {"status": JobStatus.RUNNING}) for w in newly_running]
+            )
+            for w in newly_running:
+                self.log.info("Worker %s is running", w.name)
+
+        self.db.update_workers([(w, {"close_expected": True}) for w in close_expected])
+
+
+async def _background_task(backend: OsparcBackend, cluster: Cluster):
+    backend.log.info("%s", f"{cluster.options=}")
+    while await asyncio.sleep(2, result=True):
+        backend.log.debug("osparc bck task: getting cluster information")
+        cluster_info = await get_cluster_information(backend.docker_client)
+        backend.log.debug("osparc bck task: %s", f"{cluster_info.json(indent=2)}")
+        backend.config.ClusterConfig.cluster_max_workers = len(cluster_info)
+        # backend.log.debug(
```

Comment:
MINOR: remove or annotate comments

Reply:
added a todo... that is actually part of what I want to discuss